src/applications/gi/components/profile/SchoolLocations.jsx

```diff
@@ -52,8 +55,30 @@ export class SchoolLocations extends React.Component {
     return <a href={`${facilityCode}${query}`}>{name}</a>;
   };
 
-  handleViewMoreClicked = () => {
-    this.setState({ viewMore: true });
+  handleViewAllClicked = () => {
+    this.setState({
+      viewableRowCount: this.state.totalRowCount,
+      viewAll: true,
+    });
+  };
+
+  handleViewLessClicked = () => {
+    this.setState({
+      viewableRowCount: this.state.initialRowCount,
+      viewAll: false,
+    });
+  };
+
+  showMoreClicked = () => {
+    const remainingRowCount =
+      this.state.totalRowCount - this.state.viewableRowCount;
+    if (remainingRowCount >= 10) {
+      this.setState({ viewableRowCount: this.state.viewableRowCount + 10 });
```

Comment:
Please create a constant for the magic numbers.

Reply:
Done!